src/Text/Parsing/StringParser/String.purs

```diff
@@ -46,3 +56,35 @@ string :: String -> Parser String
 string nt = Parser (\s fc sc -> case s of
   { str = str, pos = i } | indexOf' nt i str == Just i -> sc nt { str: str, pos: i + length nt }
   { pos = i } -> fc i (ParseError $ "Expected '" ++ nt ++ "'."))
+
+-- | Match a character satisfying the given predicate.
+satisfy :: (Char -> Boolean) -> Parser Char
+satisfy f = try do
+  c <- anyChar
+  if f c
+     then return c
+     else fail "Character did not satisfy predicate"
+
+-- | Match the specified character.
+char :: Char -> Parser Char
+char c = satisfy (== c)
```

Comment:
How about this:
```purescript
fail $ "Character " <> toString c <> " did not satisfy predicate"
```
I'm not sure what other information is possible to give other than the character.